File: solutions/julia/scrabble-score/2/scrabble-score.jl
scores = Dict()
for (nu,letters) in [(1,"AEIOULNRST"),(2,"DG"),(3,"BCMP"),(4,"FHVWY"),(5,"K"),(8,"JX"),(10,"QZ")]
    for ltr in letters
        scores[ltr]=nu
    end
end
function score(str)
    mapreduce((chr -> get(scores,uppercase(chr),0)),+,str;init=0)
end

File: solutions/dart/scrabble-score/1/lib/scrabble_score.dart
const scores = {
  1: 'aeioulnrst',
  2: 'dg',
  3: 'bcmp',
  4: 'fhvwy',
  5: 'k',
  8: 'jx',
  10: 'qz',
};

int score(String word) {
  int sc = 0;
  void score_letter(String chr) {
    for (final key in scores.keys) {
      if (scores[key]!.contains(chr)) {
        sc += key;
        break;
      }
    }
}
  for (final chr in word.toLowerCase().split('')) {
    score_letter(chr);
  }
    
  return sc;
}